ros/processor/suggestions_engine.py

```diff
@@ -64,25 +93,125 @@ def run(self):
             self.consumer.close()
 
 
-def download_and_extract(service, event, archive_URL, host, org_id):
-    logging.info(f"{service} - {event} - Downloading the report for system {host.get('id')}.")
+def find_root_directory(directory, target_file):
+    """
+    Recursively search for the target file in the given directory - to get root directory of an archive.
+    """
+    for dirpath, dirnames, filenames in os.walk(directory):
+        if target_file in filenames:
+            return dirpath
+
+    return None
+
+
+def get_index_file_path(service, event, host, extracted_dir):
+    extracted_dir_root = find_root_directory(extracted_dir, "insights_archive.txt")
+
+    if not extracted_dir_root:
+        logging.error(
+            f"{service} - {event} - insights_archive.txt not found in the extracted dir for system {host.get('id')}."
+        )
+        return None
+
+    pmlogger_dir = os.path.join(extracted_dir_root, "data/var/log/pcp/pmlogger/")
+
+    index_files = [
+        file for file in os.listdir(pmlogger_dir) if file.endswith(".index")
+    ]
+    index_file_path = os.path.join(pmlogger_dir, index_files[0])
```

Comment:
Maybe we can avoid this if we create named files when executing the `pmlogextract`,
~~~
$ pmlogextract -c pcp-extract-config /var/log/pcp/pmlogger/fedora/20250127.index /tmp/pmlogextract/20250127
$ ls /tmp/pmlogextract/20250127.*
/tmp/pmlogextract/20250127.0  /tmp/pmlogextract/20250127.index  /tmp/pmlogextract/20250127.meta
~~~

Reply:
I think you overlooked this part. Here, we want to get the `index` file from the extracted archive itself. We are not responsible for naming those files.